src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_SIZE = 10;
+    private static final String OF_ABSENT_MESSAGE = "Element of ArrayList ain't present";
+    private T[] storage;
+    private int size;
+
+    public ArrayList() {
+        this.storage = (T[]) new Object[START_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (storage.length == size) {
+            increaseSize();
+        }
+        storage[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        inspectIndex(index, size + 1);
+        if (storage.length < size + 1) {
+            increaseSize();
+        }
+        System.arraycopy(storage, index, storage, index + 1, size - index);
+        storage[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list != null) {
+            for (int i = 0; i < list.size(); i++) {
+                add(list.get(i));
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        inspectIndex(index, size);
+        return storage[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        inspectIndex(index, size);
+        storage[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        inspectIndex(index, size);
+        T result = storage[index];
+        System.arraycopy(storage, index + 1, storage, index, size - index - 1);
+        storage[--size] = null;
+        return result;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(storage[i], element)) {
```

Comment:
You shouldn't use Objects class in this task as described in common mistakes

Reply:
refactored